typescript/infra/scripts/transfer-ownership.ts

```diff
@@ -0,0 +1,42 @@
+import { AbacusCore, AbacusCoreChecker } from '@abacus-network/sdk';
+
+import { getCoreEnvironmentConfig, getEnvironment } from './utils';
+
+async function check() {
+  const environment = await getEnvironment();
+  const config = getCoreEnvironmentConfig(environment);
+  const multiProvider = await config.getMultiProvider();
+
+  // environments union doesn't work well with typescript
+  const core = AbacusCore.fromEnvironment(environment, multiProvider as any);
+
+  const coreChecker = new AbacusCoreChecker<any>(
+    multiProvider,
+    core,
+    config.core,
+  );
+  await coreChecker.check();
+  // 16 ownable contracts per chain.
+  await coreChecker.expectViolations(['Ownable'], [7 * 16]);
+
+  for (const violation of coreChecker.violations) {
+    const chainConnection = multiProvider.getChainConnection(violation.chain);
+    switch (violation.type) {
+      case 'Ownable': {
+        console.log(
+          `${violation.chain}: transferring ownership of ${violation.data.contract.address} from ${violation.actual} to ${violation.expected}`,
+        );
+        const response = await violation.data.contract.transferOwnership(
```

Comment:
I think there is a `transferOwnership` on the core deployer
increasingly unsure what the boundary between checker and deployer should be

Reply:
Here we have `AbacusCore`, not `AbacusCoreDeployer`. Ideally we'd have some class that's shared by both of them that we could put `transferOwnership` in, but that doesn't seem to be the case.

Only thing I can think of would be a function that takes a CoreContracts and transfers ownership.. just not sure where to put it.

Reply:
Not doing this as the AbacusCoreDeployer needs wholesale ownership transfer whereas for this we need to transfer ownership on a per-contract basis.